### DISCUSSION TITLE
Input Power vs Active Power

### DISCUSSION THREAD

**Discussion by Roving-Ronin** [JensenNick/huawei_solar_pees#5] [Forum]
Hi @JensenNick 

Sorry just getting a few minutes too look through this, as you asked for in WLCRS issues. Issues I've noted:

a) Its too expensive for a LUNA ESS in Australia, so I can't validate half of this :-(

b) You seem to be using Input Power instead of Active Power in some of your sensors. Input power is the DC power as it comes in from the PV panels, whilst Active Power is the AC Power that the inverter is outputting back into the Home (or exporting to the Grid is you have surplus). i.e. The Input Power is DC that goes into the inverter and there's a bit lost in the conversion from DC to AC, so you using Input Power will be slightly off as it isn't reflecting the 'Efficiency' %.

Note the LUNA ESS is a DC battery, so the inverter will take the PV panel (Input Power) thats DC and send it directly to the DC LUNA ESS, this is also how you can have say 10kW inverter and 15kW of PV panels and an ESS and NOT clip during summer (i.e. PV panels producing 15kW, Home using or exporting 10kW and 10/15kW ESS charging at 5kW also).
Conversely when you go to pull the kWh out of the ESS its in DC, and at this point the inverter will need to convert it into AC power. 

```
    - name: "Power House Load"
      unique_id: power_house_load
      unit_of_measurement: W
      device_class: power
      state_class: measurement
      state: >
        {{ (
        states('sensor.inverter_input_power_1') | float(0) +
        states('sensor.inverter_input_power_2') | float(0) -
        states('sensor.power_meter_active_power_1') | float(0) -
        states('sensor.battery_charge_discharge_power_1') | float(0)
        , 0 ) | max }}
    - name: "Power House Load Yield"
      unique_id: power_house_load_yield
      unit_of_measurement: W
      device_class: power
      state_class: measurement
      state: >
        {% if states('sensor.power_house_load') | float(0) >
        ( states('sensor.inverter_input_power_1') | float(0) +
        states('sensor.inverter_input_power_2') | float(0) ) %}
          {{ states('sensor.inverter_input_power_1') | float(0) +
          states('sensor.inverter_input_power_2') | float(0) }}
        {% else %}
          {{ states('sensor.power_house_load') | float(0) }}
        {% endif %}
```

**Comment by Roving-Ronin** [comment]
@JensenNick sorry I can't find the original repository we where chatting upon... was it WCLRS or another ?

Anyhow just wondering if you've seen this part of the WLCRS Wiki that covers additional dashboards 
https://github.com/wlcrs/huawei_solar/wiki/Dashboards

..that also links to https://github.com/heinemannj/home-assistant-config/ that has I believe has a number of the same sensors already defined, also might be handy for validating sensor state formulas with?

and Visualisations:
https://github.com/wlcrs/huawei_solar/wiki/Visualisations

**Comment by JensenNick** [comment]
Hi @Roving-Ronin 
Thank you for your feedback and input. This time sorry for my late reply. Yes LUNAs are really expensive and not that impressive in terms of either years or cycles warranty. I could write a whole post about that (actually I have, on the danish Huawei FB forum). Bottom line is I believe their batteries to be some of the best and the limited warranty is more of a careful business policy. Next gen batteries are 6000 cycles and 15 years warranty.

Ad b) There are some huge fundamental differences between active power and input power. Take a look at the graphs below. With my approach I couldn’t figure out how to use active power. I’m fully aware that the input power sensor is “brutto” and that there is a loss. But the loss is much less than the loss in the battery. Also, all the sensors add up. So there is no loss that hasn’t been accounted for. I believe the loss (apart from the loss in the battery) is part of the house consumption and you could off course argue that it’s not fair to include this in the consumption when I calculate the economy without pv.
 
![D2297A72-E0B7-430C-B630-8B4FBFFE9859](https://github.com/JensenNick/huawei_solar_pees/assets/149193835/a0b48466-4a7d-4f71-b7e3-bdd585323689)

I did try to use the input power sensors with loss, created according to the wiki, in the beginning. But I got some unwanted/unrealistic results. Don't quite recall what it was - but maybe something like a negative consumption. But then again I later realized the my settings were to conservative. I think I'll have a look at this again.

I’m fully aware of the conversion loss in the battery. Actually I have more sensors to share. One of them being battery efficiency. I can pretty comfortably say that the efficiency of charging from grid is aprox. 87%. The efficiency of a discharging from battery is aprox. 93% resulting in ACtoDCtoAC efficiency of 80%. Appart from this there is also a loss/ consumption when the battery is in standby - this is another post :-).

![7487DD72-DF3A-47DF-BB77-1422847AE641](https://github.com/JensenNick/huawei_solar_pees/assets/149193835/613da8c1-7b14-4bc2-9b6c-d3f1b03fe5b9)
